update slido function and documentation

kweav opened the pull request:
get_slido has been failing for me, and when @carriewright11 and I walked through the function together in coworking, we found out that the issue was with `googledrive::drive_ls` which seems to be a documented issue that's been open for some time (who knows why we're just now experiencing problems with it). Within the issue thread, we found someone's solution using `googledrive::drive_find` that's working for us, so I'm adding that change here.

New function documentation: https://www.rdocumentation.org/packages/googledrive/versions/2.0.0/topics/drive_find

Adjusted some arguments to control the behavior and give users flexibility.
- The recursive argument is no longer necessary/relevant, so I removed it
- Rather than a drive ID, we want to use a shared drive ID (or name (e.g., "ITCR")), and it will search that whole drive (hence why recursive is not relevant), so I updated that parameter name/description and how it's used ever so slightly
- I've added two arguments:
  - file_type: drive_find wants a file type, and by default I've got it set to spreadsheet which makes the most sense for slido data, but allowing this to be changed if need be
  - tags_to_find: regular expression or pattern of file names to match/find. By default, I'm selecting two file name beginners that we care most about. I give the user the flexibility to add more tags here and I keep the list of tags within the function for later data wrangling steps unchanged. 

I've tested this locally and will install from this branch to test within my data analysis to confirm that it's working how I expect it to.

After making the changes to the code, I used devtools to run `document()` to update NAMESPACE and man files since I adjusted the header and importFrom statement for googledrive. It also updated a github function man file where it seems I added an extra example of way to use the function (since if I remember correctly, I had used that function with that argument incorrectly for a dashboard and so was probably adding notes for myself but good to have it for everyone too)

googledrive Issue discussion with solution: https://github.com/tidyverse/googledrive/issues/154#issuecomment-320471321

kweav commented:
I've got why it works locally but not here -- I need to update the scopes for authorize google
<img width="1060" height="31" alt="image" src="https://github.com/user-attachments/assets/685caef2-33d4-4a90-affa-2901c6f5fac7" />

This is what I think I need to add, but it's listed as "restricted". 

I would add that scope here: https://github.com/ottrproject/metricminer/blob/fixSlido/R/scopes.R

Is that ok @carriewright11 @cansavvy?

cansavvy commented:
> I think it is great you figured out a way to make this work! I am rethinking this a tad... since we have survey data, this might not be optima, since it opens up viewing by a third party. Do you think a workaround would be possible?

Google doesn't have access but they do faciliate others giving metricminer permission to have access.

cansavvy commented:
> You'll have to authenticate to be granted the scope to search the shared drive -- is that still not optimal?

You're correct in that this would probably fix this immediate problem. But it will create a new problem. Whenever you use a restricted scope Google will have to review our app and documentation and will want a justification of why and if there's anything we could do instead to get around this. They will also require a new video to be made to show why we are using the scopes.

kweav commented:
@carriewright11 and @cansavvy I've incorporated option 3 following Candace's instructions and tested it locally and it works! I've updated the documentation as well. Note the major changes are users will have to auth twice (or select their pre-authorized account twice, once for google drive and once for google sheets -- not that bad honestly) and it does not work for me if I provide a URL or part of a URL, but does work if I use the shared drive name. 

Good to merge?

cansavvy reviewed (approved):
Seems right to me. And looks like tests are passing.
